### PR TITLE
e2e, nonroot: Execute test

### PR DESCRIPTION
**Pull request by orelmisan**
**What this PR does / why we need it**:
Currently, the `[sig-compute]NonRoot feature` test is skipped in the CI.

It is skipped because of a condition defined at `automation/test.sh` is not met since kubevirt/project-infra#2418.

Update the condition to check if the Root feature gate is disabled. The said test will run if the Root feature gate is disabled.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: automation/test.sh [hunk 1]
```diff
@@ -432,7 +432,7 @@ if [[ -z ${KUBEVIRT_E2E_FOCUS} && -z ${KUBEVIRT_E2E_SKIP} ]]; then
   fi
 fi
 
-if [[ $KUBEVIRT_NONROOT =~ true ]]; then
+if [[ $FEATURE_GATES != *"Root"* ]]; then
```
Comment (fossedihelm) [inline automation/test.sh:435]: nit: I know that `NonRoot` is the default behavior, so it is not mandatory to set the FG.
But this chek will skip non root tests if someone will use 
`export FEATURE_GATES=NonRoot`

It's probably not worth considering.

**Review by fossedihelm — COMMENTED** [review]
Thanks @orelmisan!
One little corner case that maybe it's not worth considering.
/lgtm

File: automation/test.sh (hunk 1 above)
Comment (orelmisan) [reply automation/test.sh:435]: Saying that, good catch on the pattern matching ;)

Comment (xpivarc) [inline automation/test.sh:435]: Can't we drop `(verify-non-root)` and `decorators.NonRoot`?

Comment (orelmisan) [reply automation/test.sh:435]: We sure can.
I think the meanings will be:
1. The condition could be removed from `automation/test.sh`.
2. The nonroot test should have a skip if the Root FG is enabled, so it won't fail on rootful lanes.
@xpivarc WDYT?

Comment (xpivarc) [reply automation/test.sh:435]: For the 2nd point, I would let them fail. Users can skip them. This is IMHO much better than a silent skip

Comment (xpivarc) [reply automation/test.sh:435]: What decorator are we talking about? But I agree skipping on the label is better.

Comment (fossedihelm) [reply automation/test.sh:435]: The `decorators.NonRoot` one! I think we still need it to exclude the non root tests from the root lanes

Comment (xpivarc) [reply automation/test.sh:435]: Oh, yes, 100%

Comment (orelmisan) [reply automation/test.sh:435]: Please note that this decorator is only used once, in this test.

Comment (fossedihelm) [reply automation/test.sh:435]: Yes, in the `automation/test.sh` script we only use the `--label-filter` ginkgo arg.
So without it we are not able to skip it, except with the explicit `Skip` in the code.

Comment (orelmisan) [reply automation/test.sh:435]: @fossedihelm @xpivarc I think we can remove the decorator if it is not used anywhere else.
The alternative to removing the label is a replacing the assertion with a skip in the code, which we agree is not a good practice.
Do you agree?

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/88137083139c202bf3401caf676c5c0273b8237d..9ed4d731fdc12168221b8f537f92854d2cf02a8b): Remove `NonRoot` decorator.

**Comment by fossedihelm** [comment]
> [Change](https://github.com/kubevirt/kubevirt/compare/88137083139c202bf3401caf676c5c0273b8237d..9ed4d731fdc12168221b8f537f92854d2cf02a8b): Remove `NonRoot` decorator.

Maybe I misunderstood the previous sentence! We cannot delete the label decorator since it is used to not run those tests in root lanes

**Comment by orelmisan** [comment]
> > [Change](https://github.com/kubevirt/kubevirt/compare/88137083139c202bf3401caf676c5c0273b8237d..9ed4d731fdc12168221b8f537f92854d2cf02a8b): Remove `NonRoot` decorator.
> 
> Maybe I misunderstood the previous sentence! We cannot delete the label decorator since it is used to not run those tests in root lanes

So what are you suggesting to remove?
This part in the test's title?
`[verify-nonroot] NonRoot feature`

**Comment by fossedihelm** [comment]
> So what are you suggesting to remove? This part in the test's title? `[verify-nonroot] NonRoot feature`

My suggestion is to **NOT** remove the label and anything else.
Those are useful for the root lanes. To me we can remove the decorator only when we fully drop the `Root` FG

**Comment by orelmisan** [comment]
After an offline conversation with @xpivarc, it was decided to completely remove this test.
It will be done in PR #10615.
